### DISCUSSION TITLE
Unable to login

### DISCUSSION THREAD

**Discussion by Kandaroh** [requarks/wiki#5416] [Help / Questions] [answered]
Hello, 
I am using wiki.js via docker. I followed the guide and everything was OK until a couple of days ago. For some reason my account disconnected, when i tried to log in again my old password wasn't working so i tried to reset it. However, when i try insert my email in the "forgot my password" section the answer i get is "The mail configuration is incomplete or invalid.". When i try to insert any other email it says "Check your emails for password reset instructions!". 

How can i solve this? is there a way to recover my password in the docker?

 Thanks in advance for your time, 

bests, 

Kandaroh

**Comment by NGPixel** [comment, marked as answer]
https://docs.requarks.io/en/troubleshooting#how-to-manually-reset-the-admin-password

You need to setup the mail settings under Administration > Mail for the reset password feature to work.

**Comment by KamranArain** [comment]
I am right now facing the same issue, I enabled the Oauth and disabled the UI of local login. Now OAuth is working and after using `wiki.example.com/login?all=1` it always says that password is incorrect even I reset the admin password from the db directly and also rest it as per document through the docker cmd. 

So now only non admin users are working and admin user is gone forever. I think I also have to delete the db and deploy it again.

**Reply by NGPixel** [reply]
That doesn't make any sense... The admin user isn't gone. Make sure you're providing a bcrypt hash when [resetting the password](https://docs.requarks.io/en/troubleshooting#how-to-manually-reset-the-admin-password) and not the password directly. Then try to login in another browser.

**Reply by KamranArain** [reply]
I did this multiple times and eventually I get tired of it and removed the db completely and reintializing everything

**Reply by sarathjasrin** [reply]
Happened to me as well

**Reply by sarathjasrin** [reply]
While copying the hash from the website it's adding extra space/newline

**Comment by hermanhel** [comment]
hey guys a reminder: if you are doing `docker exec db psql -U wiki -d wiki -c "UPDATE users SET password = 'HASH-PASSWORD' WHERE email = 'YOUR-EMAIL';"` in the wiki in bash, remember to escape all `$`, for example, do

```bash
docker exec db psql -U wiki -d wiki -c "UPDATE users SET password = '\$2a\$12\$iKy6CFVB7iGb3JqROibYN.cZ/cTyvVAQYFxP48bET849QmguNbuh2' WHERE email = 'YOUR-EMAIL';
```

or you will be setting the password to `a2.` something, as `$2`, `$1 and `$iKy6...` are all interpreted as bash environmental variables.

I thought it was some psql trigger and sweated it for 4 hours.
